modules/edu.gemini.seqexec.engine/src/main/scala/edu/gemini/seqexec/engine/package.scala

```diff
@@ -91,18 +102,18 @@ package object engine {
               // Final State
               case Some(qs: Sequence.State.Final) =>
                 putS(id)(qs) *> send(q)(finished(id))
-              // Execution completed
+              // Execution completed. Check breakpoint here
               case Some(qs) =>
-                putS(id)(qs) *> send(q)(executing(id))
+                putS(id)(qs) *> (if(qs.getCurrentBreakpoint) switch(q)(id)(SequenceState.Idle)
```

Comment:
You could create analogous function to `getCurrentBreakPoint` lifted to `Handle`, like `status` in this module I think, but if this only place where you do this, I don't think it's worth the effort.